### PR TITLE
[READY TO MERGE] rdkit upgrade

### PR DESCRIPTION
**Pull request by lilleswing**

### CODE REVIEW

**Comment by mwojcikowski** [comment]
B1 is a beta release. I would add the stable 2017.09.1

**Comment by lilleswing** [comment]
@mwojcikowski moved to stable.

**Comment by lilleswing** [comment]
P.S. this is looking to cascade into a pandas upgrade which will break pickled datasets.  This will force the next version to be a 2.0 cut.

**Comment by peastman** [comment]
Does that mean all existing datasets will become unreadable?  Or just ones that have been loaded and then pickled?

**Comment by lilleswing** [comment]
Because of joblib datasets are only readable on the (python_version, pandas_version) equality.  Rdkit relies on newer numpy which forces a pandas bump which will render all existing dc.data.Datasets unreadable.

**Comment by lilleswing** [comment]
We can create a tool which translates the datasets between versions at 2x disk space.

**Comment by Dgelemi** [comment]
If 2 set of data are created, would the most recent one be compatible with python 3.x ? That would also solves other issues that are encountered when trying to run benchmarks

**Comment by peastman** [comment]
I'd call that a showstopper.  We can't go around causing data to become unreadable.

If our existing dataset format is that brittle, we need to create a new one ASAP that we can guarantee permanent forward compatibility on.  And then we need to carefully consider a migration plan, including giving people lots of time to migrate their data to the new format.  That means we don't break the old format until after (preferably well after) the new format has been released.

**Comment by lilleswing** [comment]
@peastman I looked through it more and we can lazily move the metadata storage (the non-compatable blocking piece) to a more stable storage format (hd5).  

This CR accomplishes that.
https://github.com/deepchem/deepchem/pull/899
And will continue to read older DiskDatasets.  I also started a issue to track deprecating metadata joblib files.
https://github.com/deepchem/deepchem/issues/900

File: deepchem/feat/tests/test_basic.py [hunk 1]
```diff
@@ -45,6 +47,9 @@ def testRDKitDescriptors(self):
     Test simple descriptors.
     """
     descriptors = self.engine([self.mol])
+    print(descriptors)
```
Comment (rbharath) [inline deepchem/feat/tests/test_basic.py]: Perhaps remove these print statements to keep logs clean.

Comment (lilleswing) [reply deepchem/feat/tests/test_basic.py]: sure -- p.s. this was a real bug that I uncovered because of this test.

**Comment by lilleswing** [comment]
So the issues here are with travis OS which uses ubuntu 14.04 which limits the number of static TLS accesses. 

The limit is raised in 16.04

**Comment by lilleswing** [comment]
I'm really not sure how I can solve this problem.  This is a configuration in the linux kernel so using travis-ci docker will not resolve it.

I think the new tf build just uses more of the reserved static tls opens

**Comment by lilleswing** [comment]
When should we pull the trigger on this?

**Comment by rbharath** [comment]
If everyone is OK, I think we should pull the trigger now. We're in the middle of a bunch of major changes for 2.0 anyways, so this is a good spot.

**Comment by lilleswing** [comment]
Yeah gogo.

**Comment by rbharath** [comment]
Alright, pulling the trigger.

LGTM

**Pull request merged**